### PR TITLE
HLS integration testing and refactor

### PR DESCRIPTION
**Pull request by DuckyCB**
## Description
Please describe the changes.
If it's a feature or enhancement please be as detailed as possible.
If it's a bug fix, please link the issue that it fixes or describe the bug in as much detail.

## Requirements Checklist
- [ ] Unit Tests updated or fixed
- [ ] Docs/guides updated

### CODE REVIEW

File: lib/src/cmaf/ham/mapper/hls/mapHamToHls.ts [hunk 1]
```diff
@@ -1,45 +1,57 @@
...
-	NEW_LINE,
-	WHITE_SPACE,
-	WHITE_SPACE_ENCODED,
-	AT_SEPARATOR,
-	HYPHEN_MINUS_SEPARATOR,
-} from '../../../utils/constants.js';
+import { getPlaylistData, getSegments } from './utilsHamToHls.js';
+
+type ManifestPlaylistPiece = {
+	manifestToConcat: string;
```
Comment (matiasrb97) [inline lib/src/cmaf/ham/mapper/hls/mapHamToHls.ts]: We should find a better way to name these variables. It's weird to have `manifestToConcat` as part of a type. What I see, the `manifestToConcat` is the line to add in the main playlist, so maybe the name could be related to something like that.

File: lib/src/cmaf/ham/mapper/hls/mapHlsToHam.ts [hunk 2]
```diff
@@ -1,63 +1,67 @@
...
-	Track,
 	VideoTrack,
 } from '../../types/model';
 import type { Manifest, PlayList } from '../../types';
-
-function mapHlsToHam(manifest: Manifest) {
+import { addMetadataToHls } from '../../../utils/manifestUtils.js';
+import { parseHlsManifest } from '../../../utils/hls/hlsParser.js';
+import {
+	_formatSegments,
```
Comment (matiasrb97) [inline lib/src/cmaf/ham/mapper/hls/mapHlsToHam.ts]: remove the `_` in the method. We are using that for "private" functions that will not be exported.

File: lib/src/cmaf/ham/mapper/hls/mapHlsToHam.ts [hunk 3]
```diff
@@ -1,63 +1,67 @@
...
 	if (videoSwitchingSets.length > 0) {
 		selectionSets.push({
-			id: uuid(),
+			id: (selectionSetId++).toString(),
 			switchingSets: videoSwitchingSets,
 		} as SelectionSet);
 	}
 
-	return [{ id: uuid(), selectionSets: selectionSets }];
+	return [{ id: '0', selectionSets: selectionSets }];
```
Comment (matiasrb97) [inline lib/src/cmaf/ham/mapper/hls/mapHlsToHam.ts]: Why are we using '0' as id here?

File: lib/src/cmaf/ham/mapper/hls/mapHamToHls.ts [hunk 4]
```diff
@@ -1,45 +1,57 @@
...
-import {
-	NEW_LINE,
-	WHITE_SPACE,
-	WHITE_SPACE_ENCODED,
-	AT_SEPARATOR,
-	HYPHEN_MINUS_SEPARATOR,
-} from '../../../utils/constants.js';
+import { getPlaylistData, getSegments } from './utilsHamToHls.js';
+
+type ManifestPlaylistPiece = {
```
Comment (NoeliaBentancor) [inline lib/src/cmaf/ham/mapper/hls/mapHamToHls.ts:13]: Maybe this should be placed in mapper types and not in this file

File: lib/src/cmaf/ham/mapper/hls/utilsHamToHls.ts [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+	HYPHEN_MINUS_SEPARATOR,
+	WHITE_SPACE,
+	WHITE_SPACE_ENCODED,
+} from '../../../utils/constants.js';
+
+function getByterange(track: VideoTrack | AudioTrack): string {
+	if (track.byteRange) {
+		return `BYTERANGE:${track.byteRange.replace(HYPHEN_MINUS_SEPARATOR, AT_SEPARATOR)}\n`;
+	} else if (track.segments?.at(0)?.byteRange) {
+		return `BYTERANGE:0@${Number(track.segments.at(0)?.byteRange?.replace(HYPHEN_MINUS_SEPARATOR, AT_SEPARATOR).split(AT_SEPARATOR)[0]) - 1}\n`;
```
Comment (NoeliaBentancor) [inline lib/src/cmaf/ham/mapper/hls/utilsHamToHls.ts:13]:
```suggestion
		return `BYTERANGE:0${AT_SEPARATOR}${Number(track.segments.at(0)?.byteRange?.replace(HYPHEN_MINUS_SEPARATOR, AT_SEPARATOR).split(AT_SEPARATOR)[0]) - 1}\n`;
```

File: lib/src/cmaf/ham/mapper/hls/utilsHamToHls.ts [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+	}
+	return '';
+}
+
+function getPlaylistData(track: AudioTrack | VideoTrack): string {
+	return `#EXT-X-MAP:URI="${getUrlInitialization(track)}",${getByterange(track)}\n`;
```
Comment (NoeliaBentancor) [inline lib/src/cmaf/ham/mapper/hls/utilsHamToHls.ts:19]: We should export NEW_LINE from constants and use it here

File: lib/src/cmaf/ham/mapper/hls/utilsHamToHls.ts [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+}
+
+function getSegments(segments: Segment[]): string {
+	return segments
+		.map((segment: Segment): string => {
+			const byteRange: string = segment.byteRange
+				? `#EXT-X-BYTERANGE:${segment.byteRange.replace(HYPHEN_MINUS_SEPARATOR, AT_SEPARATOR)}\n`
```
Comment (NoeliaBentancor) [inline lib/src/cmaf/ham/mapper/hls/utilsHamToHls.ts:26]: Idem here, we should export NEW_LINE constant and use it here

File: lib/src/cmaf/ham/mapper/hls/utilsHamToHls.ts [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+				: '';
+			const url: string = segment.url.replaceAll(
+				WHITE_SPACE,
+				WHITE_SPACE_ENCODED,
+			);
+			return `#EXTINF:${segment.duration},\n${byteRange}\n${url}`;
```
Comment (NoeliaBentancor) [inline lib/src/cmaf/ham/mapper/hls/utilsHamToHls.ts:32]: Also here, export NEW_LINE from constants file and use it here

File: lib/src/cmaf/ham/services/converters/hlsConverter.ts [hunk 9]
```diff
@@ -1,6 +1,6 @@
 import { HlsMapper } from '../../mapper/HlsMapper.js';
 import { MapperContext } from '../../mapper/MapperContext.js';
-import type { Presentation } from '../../types/model/index.js';
+import type { Presentation } from '../../types/model';
```
Comment (NoeliaBentancor) [inline lib/src/cmaf/ham/services/converters/hlsConverter.ts:3]: Is it okay do not importing index file? I remember having troubles building the project without importing index file

**Review by matiasrb97 — APPROVED** [review]

**Pull request merged**